src/core/Expander/Expander.md

```diff
@@ -15,3 +15,74 @@ import { Expander } from 'suomifi-ui-components';
   <Expander title="Test expander 3">Test expander content 3</Expander>
 </Expander.group>;
 ```
+
+## Controlled
+
+- State for the individual Expanders are stored outside of the component and user has full control.
+- Therefore when clicking the individual Expander they are not opened by default, user have to give the logic to change it.
+- It's user's responsibility to keep the state stored outside to be updated as Open/Close All is used.
+- `defaultOpen` prop will not work when Expander is in controlled state == `open` prop is given.
+
+```jsx
+import { Expander } from 'suomifi-ui-components';
+
+const [expanderOneOpen, setExpanderOneOpen] = React.useState(false);
+const [expanderTwoOpen, setExpanderTwoOpen] = React.useState(false);
+const [expanderThreeOpen, setExpanderThreeOpen] = React.useState(
+  false
+);
+
+<>
+  <div
+    style={{
+      display: 'flex',
+      flexDirection: 'column',
+      justifyContent: 'space-between'
+    }}
+  >
+    <div
+      style={{
+        display: 'flex',
+        flexDirection: 'row',
+        justifyContent: 'space-between'
+      }}
+    >
+      <button onClick={() => setExpanderOneOpen(!expanderOneOpen)}>
+        Toggle Expander 1
+      </button>
+      <button onClick={() => setExpanderTwoOpen(!expanderTwoOpen)}>
+        Toggle Expander 2
+      </button>
+      <button
+        onClick={() => setExpanderThreeOpen(!expanderThreeOpen)}
+      >
+        Toggle Expander 3
+      </button>
+    </div>
+  </div>
+
+  <Expander.group OpenAll="Open all" CloseAll="Close all">
+    <Expander
+      title="Test expander 1"
+      open={expanderOneOpen}
+      onClick={() => setExpanderOneOpen(!expanderOneOpen)}
+    >
+      Test expander content 1
+    </Expander>
+    <Expander
+      title="Test expander 2"
+      open={expanderTwoOpen}
+      onClick={() => setExpanderTwoOpen(!expanderTwoOpen)}
+    >
+      Test expander content 2
+    </Expander>
+    <Expander
+      title="Test expander 3"
+      open={expanderThreeOpen}
+      onClick={() => setExpanderThreeOpen(!expanderThreeOpen)}
```

Comment:
Could this example be simplified and aligned with controlled examples for other components? E.g. add window.confirm for one of the expanders and have others just change state when controlled like it is now. Or keep one button in addition to these above... and make it suomifi-button!

Reply:
- Open/Close all with `window.confirm` didn't work. It was stuck in infinite loop if user presses OK to confirm. Would work if used just for toggling one by one.

- Changed to use Suomifi Button instead of the basic HTML button.

Reply:
Why did this infinite loop occur? Sounds like there might be a corner case where this might happen by accident?

Reply:
Not yet knowing the details how `window.confirm` works like this, so it needs more investigating.

Reply:
> And the infinite loops is occurring with the window.confirm dialog itself, as that keep asking it over and over. Until user presses Cancel.

This might suggest a bug in the flow since once user clicks open all button > event is triggered in the expanders > controlled expander calls onClick > user clicks ok > expander group receives the new state through provider > **expander group should update it's internal state and do nothing more**

The last step might be failing and triggering a recurring update in the controlled component since onClick is called again?

Reply:
I found the problem that caused it, it was that the function version of the setState was not used when depending on previous state.
This is now fixed and using `window.confirm`s now on the controlled example.